core/src/translate/mod.rs

```diff
@@ -97,30 +97,72 @@ pub fn translate(sql_statement: &SqlStatement) -> Result<Statement> {
             comment,
             ..
         } => {
-            let columns = columns
+            let mut primary_key: Option<Vec<usize>> = None;
+
+            let translated_columns = columns
                 .iter()
-                .map(translate_column_def)
+                .enumerate()
+                .map(|(index, column)| {
+                    let (translated_column, is_primary) = translate_column_def(column)?;
+                    if is_primary {
+                        match primary_key.as_mut() {
+                            Some(_) => {
+                                return Err(TranslateError::MultiplePrimaryKeyNotSupported.into())
+                            }
+                            None => {
+                                primary_key.get_or_insert_with(Vec::new).push(index);
+                            }
+                        }
+                    }
+                    Ok(translated_column)
+                })
                 .collect::<Result<Vec<_>>>()?;
 
-            let columns = (!columns.is_empty()).then_some(columns);
-
-            let name = translate_object_name(name)?;
+            let mut foreign_keys = Vec::new();
+
+            for constraint in constraints {
+                if let sqlparser::ast::TableConstraint::PrimaryKey {
+                    columns: primary_key_columns,
+                    ..
+                } = constraint
+                {
+                    match primary_key.as_mut() {
+                        Some(_) => {
+                            return Err(TranslateError::MultiplePrimaryKeyNotSupported.into())
+                        }
+                        None => {
+                            for column in primary_key_columns {
+                                primary_key.get_or_insert_with(Vec::new).push(
+                                    translated_columns
+                                        .iter()
+                                        .position(|v| v.name == column.value)
+                                        .ok_or_else(|| {
+                                            TranslateError::ColumnNotFoundInTable(
+                                                column.to_string(),
+                                            )
+                                        })?,
+                                );
+                            }
+                        }
+                    }
+                } else {
+                    foreign_keys.push(translate_foreign_key(constraint)?);
```

Comment:
now we handle more than one constraint variant, so it would be good to change the method `translate_foreign_key` to `translate_contraint`.

Reply:
This would imply creating an enumeration of constraints. Should we do that? At this time, different constraints are handled separately.

Reply:
this is simply because the `translate_foreign_key` gets param as constraint. it is fully ok to separate it. but it would be simple to handle `translate_contraint` rather than destructuring enum variant and pass them to the seprate function.

Reply:
I have cleaned up more that area in commit 44f96e4bdbba05f211935f87685777c02f6c5b3e, I believe it should be a decent solution especially foreseeing how that match will expand in the UNIQUE and CHECK pull requests.

Reply:
Cleaned this up even more.